pkg/netdevice/vdpa.go

```diff
@@ -54,5 +58,76 @@ func GetVdpaDevice(pciAddr string) types.VdpaDevice {
 	return &vdpaDevice{
 		detailVdpaDev,
 	}
+}
+
+// vdpaInfoProvider is a DeviceInfoProvider that handles the API information of vdpa-capable devices
+type vdpaInfoProvider struct {
+	dev      types.VdpaDevice
+	vdpaType types.VdpaType
+}
+
+// NewVdpaInfoProvider returns a new InfoProvider associated with the given VDPAInfo
+func NewVdpaInfoProvider(vdpaType types.VdpaType, vdpaDev types.VdpaDevice) types.DeviceInfoProvider {
+	vdpaInfoProvider := &vdpaInfoProvider{
+		dev:      vdpaDev,
+		vdpaType: vdpaType,
+	}
+	// Still in Device construction time, so it's still not fatal if there are discrepancies
+	// between the device information and the ResourceConfig. They should be removed by
+	// selector filtering
+	if healthy, err := vdpaInfoProvider.isHealthy(); !healthy {
```

Comment:
It's not needed, however, when I read https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/7488066fa720cb32ae77dd412fa84088f3ebac10/pkg/netdevice/pciNetDevice.go#L51-L57

I thought there was an explicit decision to warn at this phase so that later, when the device gets filtered out, there is a log that tells you why.

Reply:
Thats reported at a different level (pciNetDevice creation) not InfoProvider creation.

Im OK with keeping it here if you think its cleaner.

Reply:
I think it's not a big deal to remove it. It will make the logs easier to read.

Reply:
Finally removed it.